processor/k8sprocessor/kube/client.go

```diff
@@ -279,21 +275,48 @@ func (c *WatchClient) addOrUpdatePod(pod *api_v1.Pod) {
 	} else {
 		newPod.Attributes = c.extractPodAttributes(pod)
 	}
-	c.Pods[pod.Status.PodIP] = newPod
+	if pod.Status.PodIP != "" {
+		// compare initial scheduled timestamp for existing pod and new pod with same IP
+		// and only replace old pod if scheduled time of new pod is newer? This should fix
+		// the case where scheduler has assigned the same IP to a new pod but update event for
+		// the old pod came in later.
+		func() {
+			if p, ok := c.Pods[pod.Status.PodIP]; ok {
+				if p.StartTime != nil && pod.Status.StartTime.Before(p.StartTime) {
+					return
+				}
+			}
+			c.Pods[pod.Status.PodIP] = newPod
```

Comment:
I actually discussed this with @pmatyjasek-sumo while he was working on the extension :) Having a single map and two references to the pod just makes the things easier, since this are different domains that have no risk of conflicts

Reply:
@jpkrohling as @pmm-sumo mentioned we had some discussion on that and we decided that it should be ok if 2 keys from map will point to the same reference. With this we avoid to use two loops to iterate through two lists of Pods (one by IP and second by UID)

Reply:
Sure, that's fine, but you might want to add the current size of this map in a metric somewhere: this is the kind of data that can easily explode in terms of number of elements. It's not unusual to have tens of thousands of pods in a given cluster, so, having this info would help people running the processor to identify any possible memory issue.

Reply:
Was this done? If you don't to do it in this PR, create an issue to track this.